x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable.tsx

```diff
@@ -45,12 +46,56 @@ export class SingleMetricViewerEmbeddable extends Embeddable<
   private reload$ = new Subject<void>();
   public readonly type: string = ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE;
 
+  // API
+  public readonly functionDescription: BehaviorSubject<string | undefined>;
+  public readonly jobId: BehaviorSubject<JobId | undefined>;
+  public readonly selectedDetectorIndex: BehaviorSubject<number | undefined>;
+  public readonly selectedEntities: BehaviorSubject<MlEntityField[] | undefined>;
+  public readonly title: BehaviorSubject<string | undefined>;
+
+  private apiSubscriptions = new Subscription();
+
   constructor(
     initialInput: SingleMetricViewerEmbeddableInput,
     public services: [CoreStart, MlDependencies, SingleMetricViewerServices],
     parent?: IContainer
   ) {
     super(initialInput, {} as AnomalyChartsEmbeddableOutput, parent);
+
+    this.jobId = embeddableInputToSubject<JobId, SingleMetricViewerEmbeddableInput>(
+      this.apiSubscriptions,
+      this,
+      'jobId'
+    );
+
+    this.functionDescription = embeddableInputToSubject<
```

Comment:
this prop isn't mentioned in `SingleMetricViewerComponentApi`

Reply:
Good catch - updated the type.